src/middleware/withEgressClient.js

```diff
@@ -89,8 +89,10 @@ async function record (space, resource, bytes, servedAt, env, ctx) {
     nb: {
       resource,
       bytes,
-      servedAt: Math.floor(servedAt.getTime() / 1000)
+      servedAt: servedAt.getTime()
     },
+    expiration: Infinity, // Don't expire the invocation, so we can record egress any time
+    nonce: process.hrtime().toString(),
```

Comment:
If the partition key is now milliseconds precision we should probably also use that here right? At the moment we'd end up with multiple invocations recorded but only a single row in the DB (in the same millisecond). Using `Date.now()` here would mean we'd see the same number of invocations as DB rows (i.e. one in the same millisecond).

Reply:
@alanshaw I'm not sure I fully understand. My assumption was that in the context of generating a nonce, `process.hrtime()` would provide a more precise and unique value, particularly in a high-frequency execution environment. Does it need to align with the `servedAt` field (using Date.now())? I guess they intended to function independently, no?